x-pack/plugins/data_enhanced/public/search/search_interceptor.ts

```diff
@@ -84,16 +131,31 @@ export class EnhancedSearchInterceptor extends SearchInterceptor {
       if (id && !isSavedToBackground) this.deps.http.delete(`/internal/search/${strategy}/${id}`);
     });
 
-    return pollSearch(search, cancel, { ...options, abortSignal: combinedSignal }).pipe(
-      tap((response) => (id = response.id)),
-      catchError((e: Error) => {
-        cancel();
-        return throwError(this.handleSearchError(e, timeoutSignal, options));
+    return this.createRequestHash$(request, options).pipe(
+      switchMap((requestHash) => {
+        const cached = requestHash ? this.responseCache.get(requestHash) : undefined;
+        if (cached) {
+          this.touchCachedResponse(requestHash!);
+          return of(cached.response);
+        }
+
+        return pollSearch(search, cancel, { ...options, abortSignal: combinedSignal }).pipe(
```

Comment:
should we rather put this observable in the cache, thus allowing us to do only a single request, when we would normally send a batch of exactly the same requests ? (so we cache a promise of response, not actual response) ? 

That would be needed if we wanted to optimize canvas or dashboard experience, and probably wouldn't increase complexity (much):

https://github.com/elastic/kibana/issues/88790